bundles/org.openhab.binding.bluetooth.daikinmadoka/src/main/java/org/openhab/binding/bluetooth/daikinmadoka/internal/model/commands/GetFanspeedCommand.java

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.bluetooth.daikinmadoka.internal.model.commands;
+
+import java.util.concurrent.Executor;
+
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.bluetooth.daikinmadoka.internal.model.MadokaMessage;
+import org.openhab.binding.bluetooth.daikinmadoka.internal.model.MadokaProperties.FAN_SPEED;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ * @author blafois
+ *
+ */
+public class GetFanspeedCommand extends BRC1HCommand {
+
+    private final Logger logger = LoggerFactory.getLogger(GetFanspeedCommand.class);
+
+    private FAN_SPEED coolingFanSpeed;
+    private FAN_SPEED heatingFanSpeed;
+
+    ///////////////////////////////////////////////////////////////////////////////////////////////
+
+    @Override
+    public byte[] getRequest() {
+        return MadokaMessage.createRequest(this);
+    }
+
+    ///////////////////////////////////////////////////////////////////////////////////////////////
+
+    @Override
+    public boolean handleResponse(Executor executor, ResponseListener listener, byte @Nullable [] response) {
+
+        if (response == null) {
+            return false;
+        }
+
+        try {
+
+            MadokaMessage mm = MadokaMessage.parse(response);
+
+            this.coolingFanSpeed = FAN_SPEED.valueOf(mm.getValues().get(0x20).getRawValue()[0]);
+            this.heatingFanSpeed = FAN_SPEED.valueOf(mm.getValues().get(0x21).getRawValue()[0]);
+
+            logger.debug("coolingFanSpeed: {}", coolingFanSpeed);
+            logger.debug("heatingFanSpeed: {}", heatingFanSpeed);
+
+            listener.receivedResponse(this);
```

Comment:
The reason that I passed the executor in this `handleResponse` method was so that any eventual listener callback would be executed asynchronously on it. I had a very good reason to do it as well since the bluetooth notification callback would start dropping messages if you didn't handle the notification as quick as possible. If enough messages got dropped eventually the whole notification framework stopped sending notifications causing my binding to stop getting updates from sensors.

So in short, just make sure that any channel updates are done on a separate thread from your notifications.
```suggestion
            executor.execute(() -> listener.receivedResponse(this));
```